dashboard/app/controllers/activities_controller.rb

```diff
@@ -56,6 +56,14 @@ def milestone
       end
     end
 
+    if current_user && @script_level && @script_level.stage.lockable?
+      user_level = UserLevel.find_by(user_id: current_user.id, level_id: @script_level.level.id, script_id: @script_level.script.id)
+      # we have a lockable stage, and user_level is locked. disallow milestone requests
+      if user_level.nil? || user_level.locked?(@script_level.stage)
+        return head 403
+      end
+    end
```

Comment:
If done in `track_level_progress_sync` I think we can get into trouble if the message ends up being queued. i.e. I submit a milestone request while unlocked, the request enters the queue, the level gets locked, my message gets dequeued, and now it fails because the level is locked (even though it wasn't when I submitted).

I could move to `track_progress_for_user`, but it looks like that is only called by the milestone anyways?